### PR TITLE
feat: case insensitivity handling

### PR DESCRIPTION
**Pull request by voikin**
add checkbox to enable case-insensitivity

if this checkbox is true:
- attributes and roles are mapped **regardless of the input case**
- during the login via ldap, if we could not find such a login in the database, but we received a username from the AD server, but in a different register, then this **new username is replaced instead of the old one** for further login flow

more info in `ASP-514`

### CODE REVIEW

File: controllers/auth.go [hunk 1]
```diff
@@ -471,7 +484,11 @@ func (c *ApiController) Login() {
 				}
 			}
 
-			user, err = object.CheckUserPassword(goCtx, authForm.Organization, authForm.Username, authForm.Password, c.GetAcceptLanguage(), enableCaptcha, isSigninViaLdap, isPasswordWithLdapEnabled)
+			if user != nil && isSigninViaLdap {
```
Comment (ISulimanov) [inline controllers/auth.go:487]: let's simplify diff 

```
	var userName = authForm.Username

	if (user != nil && isSigninViaLdap) {
		userName = user.Name
	}

	user, err = object.CheckUserPassword(goCtx, authForm.Organization, userName, authForm.Password, c.GetAcceptLanguage(), enableCaptcha, isSigninViaLdap, isPasswordWithLdapEnabled)
```

File: controllers/auth.go [hunk 2]
```diff
@@ -459,10 +459,23 @@ func (c *ApiController) Login() {
 			}
 
 			if isSigninViaLdap {
-				user, err = object.GetUserByFields(authForm.Organization, authForm.Username)
+				ldap, err := object.GetLdap(authForm.LdapId)
+				if err != nil {
```
Comment (Aleksgav) [inline controllers/auth.go:463]: If we login via ldap and
- get ldap error or 
- cannot get ldap 

may be stop the flow and return error to user?

File: controllers/auth.go [hunk 3]
```diff
@@ -459,10 +459,23 @@ func (c *ApiController) Login() {
...
+					record.AddReason(fmt.Sprintf("Get Ldap error: %s", err.Error()))
+				}
+				if ldap == nil {
+					record.AddReason(fmt.Sprintf("Cannot get Ldap \"%s\"", authForm.LdapId))
+				}
+
+				if ldap != nil && ldap.EnableCaseInsensitivity {
```
Comment (Aleksgav) [inline controllers/auth.go:470]: What you think about simplify if expression
```
if ldap != nil {
 if ldap.EnableCaseInsensitivity {
  } else {
  } 
}
```
?

File: controllers/auth.go (hunk 2 above)
Comment (voikin) [reply controllers/auth.go:463]: yes, it's good idea

File: controllers/auth.go (hunk 3 above)
Comment (voikin) [reply controllers/auth.go:470]: I wanted to avoid unnecessary nesting, but maybe it would be better here

Comment (voikin) [reply controllers/auth.go]: done

File: controllers/auth.go (hunk 2 above)
Comment (voikin) [reply controllers/auth.go:463]: done

File: controllers/auth.go (hunk 1 above)
Comment (voikin) [reply controllers/auth.go:496]: done

**Review by ISulimanov — APPROVED** [review]

**Pull request merged**